pandas/core/indexers.py

```diff
@@ -133,34 +133,46 @@ def check_setitem_lengths(indexer, value, values) -> None:
 
     Returns
     -------
-    None
+    bool
+        Whether this is an empty listlike setting which is a no-op.
 
     Raises
     ------
     ValueError
         When the indexer is an ndarray or list and the lengths don't match.
     """
-    # boolean with truth values == len of the value is ok too
+    no_op = False
+
     if isinstance(indexer, (np.ndarray, list)):
-        if is_list_like(value) and len(indexer) != len(value):
-            if not (
-                isinstance(indexer, np.ndarray)
-                and indexer.dtype == np.bool_
-                and len(indexer[indexer]) == len(value)
-            ):
-                raise ValueError(
-                    "cannot set using a list-like indexer "
-                    "with a different length than the value"
-                )
+        # We can ignore other listlikes becasue they are either
+        #  a) not necessarily 1-D indexers, e.g. tuple
+        #  b) boolean indexers e.g. BoolArray
+        if is_list_like(value):
+            if len(indexer) != len(value):
+                # boolean with truth values == len of the value is ok too
+                if not (
+                    isinstance(indexer, np.ndarray)
+                    and indexer.dtype == np.bool_
+                    and len(indexer[indexer]) == len(value)
+                ):
+                    raise ValueError(
+                        "cannot set using a list-like indexer "
+                        "with a different length than the value"
+                    )
+            if len(indexer) == 0:
+                no_op = True
 
     elif isinstance(indexer, slice):
-        # slice
-        if is_list_like(value) and len(values):
+        if is_list_like(value):
             if len(value) != length_of_indexer(indexer, values):
                 raise ValueError(
                     "cannot set using a slice indexer with a "
                     "different length than the value"
                 )
+            if len(value) == 0:
```

Comment:
use not len(value) (and same as above)

Reply:
Personally I find `if len(value) == 0:` more readable / clearer about the intent ...

Reply:
its not idiomatic python (though i tend to agree), either way as long as we are consistent, and unfortunately this is not consistent with the rest of the code base